packages/flame_network_assets/lib/src/flame_network_assets.dart

```diff
@@ -0,0 +1,194 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flame/cache.dart';
+import 'package:flame_network_assets/flame_network_assets.dart';
+import 'package:flutter/foundation.dart';
+import 'package:http/http.dart' as http;
+import 'package:path/path.dart' as path;
+import 'package:path_provider/path_provider.dart';
+
+/// Function signature used by Flame Network Assets to fetch assets.
+typedef GetAssetFunction = Future<FlameAssetResponse> Function(
+  String url, {
+  Map<String, String>? headers,
+});
+
+/// Function signature used by Flame Network Assets to decode assets from a
+/// raw format.
+typedef DecodeAssetFunction<T> = Future<T> Function(Uint8List);
+
+/// Function signature used by Flame Network Assets to encode assets to a
+/// raw format.
+typedef EncodeAssetFunction<T> = Future<Uint8List> Function(T);
+
+/// Function signature by Flame Network Assets to get the app directory
+/// which is used for the local storage caching.
+typedef GetAppDirectoryFunction = Future<Directory> Function();
+
+/// {@template flame_network_assets}
+///
+/// [FlameNetworkAssets] is a class similar to Flame's assets classes (like
+/// [Images] for example), but instead of loading assets from the assets bundle,
+/// it loads from networks urls.
+///
+/// By default, [FlameNetworkAssets] uses [http.get] method to make the requests
+/// ,that can be customized by passing a different [GetAssetFunction] to the get
+/// argument on the constructor.
```

Comment:
```suggestion
/// By default, [FlameNetworkAssets] uses the [http.get] method to make the
/// requests. It can be customized by passing a different [GetAssetFunction] to
/// the `get` argument on the constructor.
```